package/MDAnalysis/analysis/align.py

```diff
@@ -232,18 +235,58 @@ def rotation_matrix(a, b, weights=None):
     :func:`rmsd` calculates the RMSD between *a* and *b*; for fitting a whole
     trajectory it is more efficient to use :func:`rms_fit_trj`. A complete fit
     of two structures can be done with :func:`alignto`. """
-    if weights is not None:
-        # weights are constructed as relative to the mean
-        weights = np.asarray(weights, dtype=np.float64) / np.mean(weights)
 
     a = np.asarray(a, dtype=np.float64)
     b = np.asarray(b, dtype=np.float64)
     N = b.shape[0]
 
+    if weights is not None:
+        weights = np.asarray(weights, dtype=np.float64)
+
     rot = np.zeros(9, dtype=np.float64)
     rmsd = qcp.CalcRMSDRotationalMatrix(a.T, b.T, N, rot, weights)
+
     return np.matrix(rot.reshape(3, 3)), rmsd
 
+def _fit_to(mobile_coordinates, ref_coordinates, atoms, weights=None):
+    """Perform an rmsd-fitting to determine roation matrix and align atoms
+
+    Parameters
+    ----------
+    mobile_coordinates : array
+        Coordinates of atoms to be aligned
+    ref_coordinates : array
+        Coordinates of atoms to be fit to
+    atoms : AtomGroup
+        Atoms to be translated
+    weights : array, optional
+        To be provided for optional weighted rmsd
+
+    Returns
+    -------
+    atoms
+        AtomGroup of translated and rotated atoms
+    old_rmsd
+        Previous rmsd of coordinates
+    new_rmsd of coordinates
+    """
+    if mass_weighted:
+        weights = np.asarray(ref_atoms.masses)
+        ref_com = ref_atoms.center_of_mass()
+        mobile_com = mobile_atoms.center_of_mass()
+    else:
+        weights = None
+        ref_com = ref_atoms.center_of_geometry()
+        mobile_com = mobile_atoms.center_of_geometry()
+
+    old_rmsd = rms.rmsd(mobile_coordinates, ref_coordinates)
```

Comment:
can you remove this calculation here please. I thought this was supposed to be the light weight method.